src/services/aggregatorService.ts

```diff
@@ -145,14 +150,32 @@ export default class AggregatorService {
       sourceTimeout
     );
 
-    const filteredOptions = swapOptionsResponse.filter((option) => !('failed' in option)) as QuoteResponse[];
+    // eslint-disable-next-line @typescript-eslint/no-unnecessary-type-assertion
+    const validOptions = (swapOptionsResponse as (QuoteResponse | FailedQuote)[]).reduce(
```

Comment:
there is no need for parsing the type, it should already come as `(QuoteResponse | FailedQuote)[]` which could also remove the eslint-disable :)

Reply:
btw instead of using a reduce, we could use a map, it returns null when the quote is failed, and returns the quote when it is not failed, and then you can use the `compact` function from lodash to filter out the nulls

Reply:
Sure! It's done now. I kept the `as QuoteResponse[]` that was already there, becuase it's not inferring return type from sdkService.getSwapOptions